### PR TITLE
Fix Code Sniff Action Change Detection

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We aren't currently running PHPCS due to a problem with the detection of changed files. This pull request fixes the configuration errors causing this to happen.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. I will be pushing a series of commits to this PR to test the change detection.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by ObliviousHarmony** [comment]
Here are the test runs that I've done with this change:

* No Changes: https://github.com/woocommerce/woocommerce/actions/runs/6317061721/job/17153012080?pr=40438
* Relevant Change: https://github.com/woocommerce/woocommerce/actions/runs/6316977210/job/17152727253?pr=40438
* Irrelevant Change: https://github.com/woocommerce/woocommerce/actions/runs/6317043393/job/17152949211?pr=40438

File: .github/workflows/pr-code-sniff.yml [hunk 1]
```diff
@@ -49,7 +48,4 @@ jobs:
             - name: Run PHPCS
               if: steps.changed-files.outputs.any_changed == 'true'
               working-directory: plugins/woocommerce
-              run: |
-                  HEAD_REF=$(git rev-parse HEAD)
-                  git checkout $HEAD_REF
-                  phpcs-changed -s --git --git-base ${{ github.base_ref }} ${{ steps.changed-files.outputs.all_changed_files }}
+              run: phpcs-changed -s --git --git-base ${{ github.event.pull_request.base.sha }} ${{ steps.changed-files.outputs.all_changed_files }}
```
Comment (ObliviousHarmony) [inline .github/workflows/pr-code-sniff.yml:51]: `github.base_ref` was giving `trunk` but that requires a branch that exists in the clone that doesn't exist. This uses the base branch SHA ref which is always present in this clone.

**Comment by barryhughes** [comment]
Hi @ObliviousHarmony!

- The last run of the code sniff action in this PR didn't run properly: https://github.com/woocommerce/woocommerce/actions/runs/6317061721/job/17153012080?pr=40438
- Same thing can be seen in a test PR I created: https://github.com/woocommerce/woocommerce/actions/runs/6317232994/job/17153562587?pr=40445

<table><tr><td>Screenshot 2023-09-26 at 12 19 08</td><td>Screenshot 2023-09-26 at 12 19 15</td></tr></table>

So we're basically back at the earlier phase of the problem (it looks like linting passed, unless you take the time to click through and see the details). Or am I misunderstanding, and the goal here is just to stop this required check from blocking PRs until we arrive at a substantive fix? 🤔

**Comment by barryhughes** [comment]
OK, I misunderstood (thanks for clarifying!). With a 'bad' PHP change in place, this is doing it's thing: https://github.com/woocommerce/woocommerce/actions/runs/6317423787/job/17154179598?pr=40445

**Review by barryhughes — APPROVED** [review]
LGTM!

**Pull request merged**